### DISCUSSION TITLE
Higher Order Component

### DISCUSSION THREAD

**Discussion by corwisto** [radix-ui/themes#42] [Q&A]
Hi!

I think it could be very useful to have something like a higher order component, that could be used to encapsulate just the components you need. I wanted to have some of the theme styles, but the project I was working on uses `tailwind` and adding the theme to the root of my app was overriding all of the work we've done in the past.

if there is something like this...
`withRadixTheme(Component, RadixConfig)`
or a workaround to have this working with tailwind I would be very happy to learn more about it.

Thanks for your time reading this post!

**Comment by vladmoroz** [comment]
> the project I was working on uses tailwind and adding the theme to the root of my app was overriding all of the work we've done in the past

We aren't prioritising slotting into existing projects for now, but you might just need to customize the basic tokens to match your current style, like typography and text/background color. Mainly these:

<img width="389" alt="image" src="https://github.com/radix-ui/themes/assets/8441036/f232eaae-c554-446d-a10b-d00104a7b146">
